File: BaekJoon/Silver/1018/Python/1018.py
import sys

row, column = map(int, sys.stdin.readline().split())
board = []
for i in range(0, row):
    board.append(sys.stdin.readline().rstrip())

row_pointer = 0
min = None
while row_pointer + 7 < row:
    column_pointer = 0
    while column_pointer + 7 < column:
        chessboard = []
        for i in range(row_pointer, row_pointer+8):
            chessboard.append(board[i][column_pointer:column_pointer+8])
        countVer1 = 0
        countVer2 = 0
        for i, each_row in enumerate(chessboard):
            for j, char in enumerate(each_row):
                if (i+j) % 2 == 0:
                    if char != 'W':
                        countVer1 += 1
                    else:
                        countVer2 += 1
                else:
                    if char != 'B':
                        countVer1 += 1
                    else:
                        countVer2 += 1
        count = countVer1 if countVer1 < countVer2 else countVer2
        if min == None:
            min = count
        else:
            if count < min:
                min = count
        column_pointer += 1
    row_pointer += 1

print(min)

File: BaekJoon/Silver/1018/C++/1018.cpp
// no.1018: 체스판 다시 칠하기 (S4)

#include <cstdio>
using namespace std;

int count(char [][8]);

int main() {
    int n, m;
    scanf("%d %d", &n, &m);
    char board[51][51];
    int min = 64;
    for(int i=0; i<n; i++) {
        scanf("%s", board[i]);
    }
    for(int i=0; i<=n-8; i++) {
        for(int j=0; j<=m-8; j++) {
            char cut[8][8];
            for(int k=0; k<8; k++) {
                for(int l=0; l<8; l++) {
                    cut[k][l] = board[i+k][j+l];
                }
            }
            int ret = count(cut);
            min = (ret < min) ? ret : min;
        }
    }
    printf("%d\n", min);
}

int count(char cut[][8]) {
    int point1 = 0;
    for(int i=0; i<8; i++) {
        for(int j=0; j<8; j++) {
            if((i+j)%2==0 && cut[i][j]!='W') {
                point1++;
            }
            if((i+j)%2!=0 && cut[i][j]!='B') {
                point1++;
            }
        }
    }
    int point2 = 0;
    for(int i=0; i<8; i++) {
        for(int j=0; j<8; j++) {
            if((i+j)%2==0 && cut[i][j]!='B') {
                point2++;
            }
            if((i+j)%2!=0 && cut[i][j]!='W') {
                point2++;
            }
        }
    }
    return (point1 < point2) ? point1 : point2;
}